src/libs/barcode/lv_barcode.c

```diff
@@ -195,43 +195,28 @@ static void lv_barcode_destructor(const lv_obj_class_t * class_p, lv_obj_t * obj
 {
     LV_UNUSED(class_p);
 
-    lv_image_dsc_t * img = lv_canvas_get_image(obj);
-    lv_cache_lock();
-    lv_cache_invalidate_by_src(img, LV_CACHE_SRC_TYPE_POINTER);
-    lv_cache_unlock();
-
-    if(!img->data) {
-        LV_LOG_INFO("canvas buffer is NULL");
-        return;
-    }
-
-    LV_LOG_INFO("free canvas buffer: %p", lv_canvas_get_buf(obj));
+    lv_draw_buf_t * draw_buf = lv_canvas_get_draw_buf(obj);
+    if(draw_buf == NULL) return;
 
-    lv_draw_buf_free((void *)lv_canvas_get_buf(obj));
-    img->data = NULL;
+    lv_draw_buf_destroy(draw_buf);
```

Comment:
Shouldn't it be handled in `lv_canvas`?

Reply:
The draw buffer is allocated in `bardcode`, thus should be destroyed in `barcode` too. Canvas could also use static draw buffer, thus cannot destroy externally provided buffer.